app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -54,6 +54,43 @@ pub struct MissingThisOnMethod(pub String);
 
 
 
+// =====================
+// === QualifiedName ===
+// =====================
+
+im_string_newtype! {
+    /// A single segment of a [`QualifiedName`] of an [`Entry`].
+    QualifiedNameSegment
+}
+
+/// A fully qualified name of an [`Entry`].
+///
+/// A string representation of a fully qualified name can be built by joining the
+/// [`QualifiedName::segments`] using an [`ast::opr::predefined::ACCESS`] character.
```

Comment:
this should be a method instead of a description in the docs how to code such a method.